### PR TITLE
Markdown template support

### PR DESCRIPTION
**Pull request by agarthetiger**
PR for preview and discussion of possible implementation of markdown support. This is not ready for merge yet.

Added .gitignore for PyCharm files, plus other files generated when using `pip install -e .` to test the code locally.
md_ify still needs transforming for all elements.
module.md.j2 included for reference, not necessarily for inclusion in final release. It would need a flag to switch between this and the default .rst template, as it would be tricky to pass the full path to this file after this package is installed via pip.

### CODE REVIEW

File: src/ansible_doc_extractor/cli.py [hunk 1]
```diff
@@ -56,27 +66,37 @@ def render_module_docs(output_folder, module, template):
...
-    env.filters["rst_ify"] = rst_ify
+    env.filters["ensure_list"] = ensure_list
     if custom_template:
+        extension = custom_template.name.split('.')[-2]
+        if extension not in _supported_templates:
+            raise AttributeError(
+                "Template type not supported. Template type must be one of "
+                "the following types: {}".format(_supported_templates)
+            )
+        env.filters[extension + "_ify"] = globals()[extension + "_ify"]
```
Comment (tadeboro) [inline src/ansible_doc_extractor/cli.py]: There is no need for this. We can always register all our custom filters no matter what kind of template user is providing us with.

File: src/ansible_doc_extractor/cli.py [hunk 2]
```diff
@@ -56,27 +66,37 @@ def render_module_docs(output_folder, module, template):
...
+        output_folder, doc["module"] + "." + extension
+    )
     with open(module_rst_path, "w") as fd:
         fd.write(template.render(doc))
 
 
 def get_template(custom_template):
     env = Environment(loader=PackageLoader(__name__), trim_blocks=True)
-    env.filters["rst_ify"] = rst_ify
+    env.filters["ensure_list"] = ensure_list
```
Comment (tadeboro) [inline src/ansible_doc_extractor/cli.py]: While I do not mind adding yet another filter into the mix, I would much rather work towards a more strict specification of what the templates receive as data. For example, making sure that all descriptions are actually lists of paragraphs, same for authors, etc. This way all the heavy lifting can be done in a programming language, not a templating language.

File: src/ansible_doc_extractor/cli.py (hunk 1 above)
Comment (agarthetiger) [reply src/ansible_doc_extractor/cli.py:86]: Fair point, I'll update to always register both.

**Comment by agarthetiger** [comment]
Apologies for the delay. A member of my family passed away last week (not covid-related). I'll pick this up again next week. Thank you for the feedback.

**Comment by tadeboro** [comment]
Please accept our deepest condolences to you and your family. And do not worry about this PR, you are not blocking anyone's work, so please, do take care of important things first.

File: src/ansible_doc_extractor/cli.py (hunk 2 above)
Comment (agarthetiger) [reply src/ansible_doc_extractor/cli.py]: I've removed this now.

**Comment by agarthetiger** [comment]
Thank you for your patience, I'm back looking at this again now. 
I've added a --markdown flag to the args as I couldn't see another way to switch to using an embedded markdown template. I see there is support for passing any other template, so I also wanted to check if you still wanted a default markdown template as part of this tool? If so, I'm happy to change how we switch between rst (still the default) and the bundled markdown template.

**Comment by faupol3** [comment]
@tadeboro any updates on reviewing this? Let me know if I can be of help, I'm interested in this feature as well

**Comment by tadeboro** [comment]
I will be honest with you all: I completely forgot about this pull request and I feel really bad for dropping the ball. I will review this PR by the end of the week. Thank you, @faupol3, for the reminder.

**Comment by SirUli** [comment]
@tadeboro any chance to get that review by the end of any week soon? Thanks in advance!

File: src/ansible_doc_extractor/templates/module.md.j2 [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+```
+
+{% endif %}
+{% macro result_desc(results, level) %}
+{%   for name, spec in results.items() %}
+{%     set ret = spec.returned %}
+{%     set typ = spec.type | default("any") %}
+{%     set sample = spec.sample %}
+  {{ "  " * level }}{{ name }} ({{ ret }}, {{ typ }}, {{ sample }})
+    {{ "  " * level }}{{ spec.description | md_ify }}
```
Comment (juremedvesek) [inline src/ansible_doc_extractor/templates/module.md.j2:92]: Spec description is a list - that change fixes it. 

```suggestion
{% for desc_line in spec.description  %}
    {{ "  " * level }}{{ desc_line | md_ify }}
{% endfor %}
```

**Review by juremedvesek — CHANGES_REQUESTED** [review]
Merge conflicts should be simple to solve .

**Comment by SirUli** [comment]
@agarthetiger could you please consider these requested changes for your code? Thanks in advance! I'd really like to see MD support soon :)

**Review by juremedvesek — APPROVED** [review]
Some polishing of template is still needed. But would be done as iterative process.

**Pull request merged**